### ISSUE TITLE
Unrecognized content encoding type. libcurl understands deflate, gzip content encodings.

### ISSUE THREAD

**Issue by MaTToX3** [guzzle/guzzle#2146]
| Q            | A
| ------------ | ---
| Bug?         | yes
| New Feature? | no|yes
| Version      | ~6.0

#### Actual Behavior

What is the actual behavior?
When sending a request to a 3rd party provider, request is accepted and succesfully ran but then error is thrown:  cURL error 61: Unrecognized content encoding type. libcurl understands deflate, gzip content encodings.

#### Expected Behavior

What is the behavior you expect?
Response status 200 should be returned and application should continue executing normally.

#### Steps to Reproduce

Send a request on APACHE server - this is not an issue on NGINX. 

Request debug output:
*   Trying 85.10.14.67...
* TCP_NODELAY set
* Connected to eurofaktura.com (85.10.14.67) port 443 (#0)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=racuni.com
*  start date: Jul 24 11:44:02 2018 GMT
*  expire date: Oct 22 11:44:02 2018 GMT
*  subjectAltName: host "eurofaktura.com" matched cert's "eurofaktura.com"
*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
*  SSL certificate verify ok.
> POST /WebServicesSI/API HTTP/1.1
Host: eurofaktura.com
User-Agent: GuzzleHttp/6.2.1 curl/7.60.0 PHP/7.1.21
Content-Type: application/xml; charset=UTF8
Content-Length: 773

* upload completely sent off: 773 out of 773 bytes
< HTTP/1.1 200 OK
< Server: nginx/1.12.1
< Date: Wed, 19 Sep 2018 07:55:42 GMT
< Content-Type: text/xml
< Content-Length: 246
< Connection: keep-alive
< X-ErApp: WebServicesSI
< Set-Cookie: ItcClientID=5WQEJYMxLJagzx4Iaibkif98_3714803742; Expires=Tue, 31-Dec-2030 23:59:59 GMT; Path=/; HttpOnly
< Content-Encoding: 8-bit
< Strict-Transport-Security: max-age=63072000
<
* Unrecognized content encoding type. libcurl understands deflate, gzip content encodings.
* stopped the pause stream!
* Closing connection 0

**Comment by eb-dteixeira** [comment]
We're experiencing exactly the same issue, but with `GuzzleHttp/6.3.3 curl/7.29.0 PHP/5.6.37` and a request returning `UTF-8` encoded data. Passing `'decode_content' => false` on the request options allows us to bypass the issue, but we'd like to keep compression handling automatic.

**Comment by sagikazarmark** [comment]
Have you tried sending plain curl requests? I suspect that curl won't work with this: `Content-Encoding: 8-bit` You should probably configure content encoding on the server side.

**Comment by MaTToX3** [comment]
@eb-dteixeira I will try, thank you for that!

@sagikazarmark I can't, server-side is 3rd party and I have already been in contact with them and they say they can't do anything about it. And as I said it, it works on my local home server, just not on production anymore. But good tip with plain CURL request, will definetly try it if the first tip does not help me!

**Comment by sagikazarmark** [comment]
@MaTToX3 if id doesn't work with curl, it probably won't work with guzzle either.

**Comment by MaTToX3** [comment]
@sagikazarmark on my local server that runs NGINX, the return Content-Encoding is also 8-bit and it works. I have no idea if that Content-Encoding was always the case. 

Also, setting 'decode_content' to false did not solve the issue. 

Any other tips are appreciated!

**Comment by oscarAG** [comment]
> We're experiencing exactly the same issue, but with `GuzzleHttp/6.3.3 curl/7.61.1 PHP/5.6.37` and a request returning `UTF-8` encoded data. Passing `'decode_content' => false` on the request options allows us to bypass the issue, but we'd like to keep compression handling automatic.

Fixed this issue for me

**Comment by brightobluefin** [comment]
if you are using vagrant machine , check your php version inside. sometimes it updates the version to 7.2 when you update or if you do fresh install.

**Comment by paulycoding** [comment]
'decode_content' => false

**Comment by paulycoding** [comment]
'decode_content' => false
指定是否自动解码内容编码响应（gzip，deflate等）。

此选项可用于控制如何处理内容编码的响应主体。 默认情况下，decode_content设置为true，这意味着Guzzle将解码任何gzip压缩或缩小的响应。

设置为false时，响应的主体永远不会被解码，这意味着字节不会通过处理程序。

**Comment by CleverMind** [comment]
Use header "accept-encoding" => "gzip, deflate"
Tried on Apache. Works fine.

**Comment by ihightower** [comment]
> Use header "accept-encoding" => "gzip, deflate"
> Tried on Apache. Works fine.

yes it worked for me too... i changed from

`'accept-encoding: gzip, deflate, br'`

to 

`'accept-encoding: gzip, deflate'`

**Comment by thisara-soori** [comment]
`'accept-encoding: gzip, deflate'` worked for me instead of `accept-encoding: gzip, deflate, br`

**Comment by sergotail** [comment]
I found that this is curl problem: https://github.com/curl/curl/pull/2298
This bug has been fixed in 7.59.0

**Comment by oussaka** [comment]
i use guzzle 7 and it works with this snippet :

```
        try {
            $response = $this->client->delete(url."/v1/webservice".$publicId",
                [
                    'headers' => $headers,
                    'decode_content' => false,
                ]
            );
        } catch (ClientException $e) {
            $response = $e->getResponse();
        }
```
